packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/prompt-helpers.ts

```diff
@@ -43,41 +43,57 @@ export async function connectWithBackendSystem(
 ): Promise<ValidationResult> {
     // Create a new connection with the selected system
     let connectValResult: ValidationResult = false;
-    if (backendSystem) {
+    let backendSystemWithCreds: BackendSystem | undefined;
+    // Fetch the credentials for the system
+    try {
+        backendSystemWithCreds = await new SystemService(LoggerHelper.logger).read(
+            BackendSystemKey.from(backendSystem) as BackendSystemKey
+        );
+    } catch (error) {
+        LoggerHelper.logger.error(t('errors.storedSystemReError', { systemName: backendSystem.name, error }));
```

Comment:
Does this occur when the user denies access, if so it should probably be a warning rather than an error?